Library/Homebrew/utils/ruby.sh

```diff
@@ -10,24 +13,9 @@ test_ruby() {
     return 1
   fi
 
-  supported_ruby_versions=()
-  if [[ -n "${HOMEBREW_RUBY3}" && -z "${HOMEBREW_USE_RUBY_FROM_PATH}" ]]
-  then
-    supported_ruby_versions+=("3.1.0")
-  fi
-  supported_ruby_versions+=("${HOMEBREW_REQUIRED_RUBY_VERSION}")
-
-  for ruby_version in "${supported_ruby_versions[@]}"
-  do
-    if "$1" --enable-frozen-string-literal --disable=gems,did_you_mean,rubyopt \
-       "${HOMEBREW_LIBRARY}/Homebrew/utils/ruby_check_version_script.rb" \
-       "${ruby_version}" 2>/dev/null
-    then
-      return 0
-    fi
-  done
-
-  return 1
+  "$1" --enable-frozen-string-literal --disable=gems,did_you_mean,rubyopt \
+    "${HOMEBREW_LIBRARY}/Homebrew/utils/ruby_check_version_script.rb" \
+    "${HOMEBREW_REQUIRED_RUBY_VERSION}" 2>/dev/null
 }
 
 can_use_ruby_from_path() {
```

Comment:
> allow this on macOS a bit more widely if we're gonna portable Ruby for everyone

This being prioritising arbitrary system Rubies or the `HOMEBREW_USE_RUBY_FROM_PATH` variable?

> Still think it'd be nice to allow the relevant Homebrew Ruby to be used if it's installed both here and for things like rubydoc.brew.sh etc.

homebrew-core Ruby is slower than Portable Ruby but could maybe visit that in a separate PR.

Reply:
> This being prioritising arbitrary system Rubies or the `HOMEBREW_USE_RUBY_FROM_PATH` variable?

I was thinking the prior but I'm very open to being talked out of that.

> homebrew-core Ruby is slower than Portable Ruby but could maybe visit that in a separate PR if there's a benefit for doing so.

Yeh, I'm thinking less about the speed etc. but more for platforms where we don't yet support portable Ruby yet and being able to more easily test new Ruby versions that way. Maybe overkill.

> A setup-ruby type action that finds and uses the Ruby that Homebrew uses could make sense rather than aligning "3.1" everywhere like we currently are. I could write something today if that sounds interesting.

This seems like a good idea.

Reply:
> more for platforms where we don't yet support portable Ruby yet and being able to more easily test new Ruby versions that way

Right now that's enabled for Linux by default. Currently prioritised over Portable Ruby, as that's how it's always been, but am tempted to switch that round.

For newer versions, you need `HOMEBREW_USE_RUBY_FROM_PATH` anyway to bypass the strict 3.1 version check. This should work on both macOS and Linux today.